frontend/src/actions/Actions.js

```diff
@@ -11,131 +11,166 @@ import {
 import { getView } from '../reducers/viewHandler';
 
 /*
- * @method getTableActions
+ * @method fetchQuickActions
  * @summary Action creator that calls the quick actions fetch internally for
  * when we're updating the table selection 
  
- * @param {string} tableId
  * @param {number} windowId
  * @param {string} viewId
  * @param {boolean} isModal
+ * @param {string} viewProfileId
  */
-export function getTableActions({ tableId, windowId, viewId, isModal }) {
+export function fetchQuickActions({
+  windowId,
+  viewId,
+  isModal,
+  viewProfileId = null,
+}) {
   return (dispatch, getState) => {
+    let actionPromises = null;
     const state = getState();
-    const table = getTable(state, tableId);
-    const selectedIds = table.selected;
-    const { includedView } = state.viewHandler;
-    const viewProfileId =
-      includedView.windowId === windowId ? includedView.viewProfileId : null;
-    let fetchActions = true;
-
-    if (viewId) {
-      const quickActionsId = getQuickActionsId({
-        windowId,
-        viewId,
-      });
-      const quickActions = getQuickActions(state, quickActionsId);
-      fetchActions = quickActions.pending === false;
-    }
+    const includedView = state.viewHandler.includedView.windowId
+      ? state.viewHandler.includedView
+      : null;
 
-    if (fetchActions) {
-      dispatch(
-        fetchQuickActions({
-          windowId,
-          viewId,
-          viewProfileId,
-          selectedIds,
-        })
-      );
+    if (includedView) {
+      viewProfileId = viewProfileId || includedView.viewProfileId;
+
+      const isParent = includedView.parentId === windowId;
+      const isChild = includedView.windowId === windowId;
+
+      if (isParent || isChild) {
+        const requests = dispatch(
+          getQuickActionRequests({
+            windowId,
+            viewId,
+            includedView,
+            viewProfileId,
+            isModal,
+            isParent,
+          })
+        );
+
+        actionPromises = [].concat(requests);
+      }
+    } else {
+      const tableId = getTableId({ windowId, viewId });
+      const table = getTable(state, tableId);
+
+      actionPromises = [
+        dispatch(
+          requestQuickActions({
+            windowId,
+            viewId,
+            selectedIds: table.selected,
+            viewProfileId,
+          })
+        ),
+      ];
     }
 
-    dispatch(
-      fetchIncludedQuickActions({
-        windowId,
-        selectedIds,
-        isModal,
-      })
-    );
+    return Promise.all(actionPromises);
   };
 }
 
 /*
  * @method fetchIncludedQuickActions
- * @summary Action creator that calls the quick actions fetch internally for parent/child
- * quick actions, when a table selection has changed
+ * @summary Action creator that calls the creates the requests for included quick actions
  *
- * @param {string} windowId
- * @param {array} selectedIds
+ * @param {number} windowId
```

Comment:
string

Reply:
windowid is a number. Viewid is a string

Reply:
IMPORTANT: pls always handle IDs are plain Strings.
This will allow the backend to change the way we handle IDs or their structure without any notice.